stl/inc/iterator

```diff
@@ -119,24 +131,28 @@ class insert_iterator { // wrap inserts into container as output iterator
 public:
     using iterator_category = output_iterator_tag;
     using value_type        = void;
-    using difference_type   = void;
     using pointer           = void;
     using reference         = void;
+    using container_type    = _Container;
 
-    using container_type = _Container;
+#ifdef __cpp_lib_concepts
+    using difference_type = ptrdiff_t;
+    insert_iterator()     = default;

```

Comment:
You are missing constexpr/_CONSTEXPR20 here.

By the way #491 hisses at you mightily

Reply:
This defaulted default constructor doesn't have an explicit `constexpr` in the working draft. I'm stomping on #491's context a bit, but I think I've managed to avoid stomping on the actual changed lines - resolving the conflicts should be trivial.